# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsHttpChannel.cpp
# Commit: afbbc75a2872
# Full Hash: afbbc75a2872b2b825d3e621dc0d398fd4c7ed34
# Author: smayya <smayya@mozilla.com>
# Date: 2025-08-08 16:22:36
# Description:
#   Bug 1980035 - add null check for mTransaction in nsHttpChannel::ProcessLNAActions. r=necko-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D260388
# ==============================================================================

diff -r 765f961d23e7 -r afbbc75a2872 netwerk/protocol/http/nsHttpChannel.cpp
--- a/netwerk/protocol/http/nsHttpChannel.cpp	Fri Aug 08 08:47:24 2025 +0000
+++ b/netwerk/protocol/http/nsHttpChannel.cpp	Fri Aug 08 08:49:07 2025 +0000
@@ -8172,6 +8172,12 @@
 }
 
 nsresult nsHttpChannel::ProcessLNAActions() {
+  if (!mTransaction) {
+    // this could happen with rcwn enabled.
+    // We have hit network and have detected LNA, meanwhile cache won and reset
+    // the transaction in ReadFromCache
+    return NS_ERROR_LOCAL_NETWORK_ACCESS_DENIED;
+  }
   // Suspend to block any notification to the channel.
   // This will get resumed in
   // nsHttpChannel::OnPermissionPromptResult
